instaclustr/structs.go

```diff
@@ -100,6 +101,11 @@ type CreateRequest struct {
 	PCICompliantCluster   string                    `json:"pciCompliantCluster,omitempty"`
 	RackAllocation        *RackAllocation           `json:"rackAllocation,omitempty"`
 	OidcProvider          string                    `json:"oidcProvider,omitempty"`
+	PrivateLink           *PrivateLinkConfig        `json:"privateLink,omitempty" mapstructure:"private_link,omitempty"`
+}
+
+type PrivateLinkConfig struct {
+	IamPrincipalARNs *[]string `json:"iamPrincipalARNs,omitempty" mapstructure:"iam_principal_arns,omitempty"`
```

Comment:
**[MINOR]** Would `IAMPrincipalARNs` be better?

Reply:
Yeah, that's better, I will change that.